syntax = "proto3";

package lkpMessage;

enum commandID{
    UPDATE = 0;      //need no parameter
    RUN = 1;         //optional: docker_num
    RESULT = 2;      //need no parameter
    PUSH = 3;        //required: testcase num and length
}

message Command {
    commandID command = 1;
    oneof testcase_oneof{
        string testcase = 2;
        string testcluster = 3;
    }
    uint32 docker_num = 4;
    string testcase_name = 5;
    uint32 testcase_len = 6;
}

message File {
    enum filetype{
        TESTCASE = 0;    //Server push testcase to client
        RESULT = 1;      //Server request result, client send to server
    }
    filetype file_type = 1;
    string file_name = 2;
    uint32 file_len = 3;
    uint32 patch_num = 4;
    bytes content = 5;
}

message CommandACK {
    bool status = 1;
    commandID command = 2;
    string ack_message = 3;  //Error message, etc.
}

message HeartBeat {
    bool status = 1;
}

message PushACK {
    bool status = 1;
    string ack_message = 2;
}